pip_check_reqs/common.py

```diff
@@ -58,18 +58,28 @@ def visit_ImportFrom(self, node: ast.ImportFrom) -> None:
     def _addModule(self, modname: str, lineno: int) -> None:
         if self._options.ignore_mods(modname):
             return
+        path_finder = importlib.machinery.PathFinder()
         path = None
         progress = []
         modpath = last_modpath = None
         for p in modname.split("."):
             try:
-                file, modpath, description = imp.find_module(p, path)
-            except ImportError:
-                # the component specified at this point is not importable
-                # (is just an attr of the module)
-                # *or* it's not actually installed, so we don't care either
+                find_spec_result = path_finder.find_spec(p, path)
```

Comment:
Quoting [`PathFinder` docs](https://docs.python.org/3/library/importlib.html#importlib.machinery.PathFinder.find_module):

Deprecated since version 3.4: Use [find_spec()](https://docs.python.org/3/library/importlib.html#importlib.machinery.PathFinder.find_spec) instead.

Reply:
I'm more than happy to ditch this - I put this up mostly to see what happened in the tests across Python versions.